doc/programming_guide/event_loop.rst

```diff
@@ -5,34 +5,154 @@ Introduction
 ------------
 
 Python Arcade provides a simple three event loop to build off.
+All three methods are exposed to be overridden in :py:class:`arcade.Window`
+and :py:class:`arcade.View`. You may also register your own handlers
+to these events using :py:func:`arcade.Window.push_handlers`, but this is
+not recommended for beginners.
```

Comment:
For advanced use cases it is possible to add own handler via :py:func:`arcade.Window.push_handlers`.